hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/impl/OperationExecutorImpl.java

```diff
@@ -112,17 +118,19 @@ public OperationExecutorImpl(GroupProperties properties,
     }
 
     private OperationRunner[] initGenericOperationRunners(GroupProperties properties, OperationRunnerFactory runnerFactory) {
-        int genericThreadCount = properties.getInteger(GENERIC_OPERATION_THREAD_COUNT);
-        if (genericThreadCount <= 0) {
+        int threadCount = properties.getInteger(GENERIC_OPERATION_THREAD_COUNT);
+        if (threadCount <= 0) {
             // default generic operation thread count
             int coreSize = Runtime.getRuntime().availableProcessors();
-            genericThreadCount = Math.max(2, coreSize / 2);
+            threadCount = Math.max(2, coreSize / 2);
         }
 
-        OperationRunner[] operationRunners = new OperationRunner[genericThreadCount];
+
+        OperationRunner[] operationRunners = new OperationRunner[threadCount + priorityThreadCount];
```

Comment:
It would be a good improvement for the properties, indeed. For now I would just add it after this block:
```
int threadCount = properties.get(...)
if (threadCount <= 0) {
  (...)
}
checkNotNegative("useful text", priorityThreadCount);
```
I think we have something in the Preconditions for this kind of check? So it should be a one-liner for now.